addons/product/models/product_pricelist.py

```diff
@@ -195,17 +197,17 @@ def _compute_price_rule(self, products, qty, uom=None, date=False):
             else:
                 # fall back on Sales Price if no rule is found
                 price = product.price_compute('list_price', uom=target_uom, date=date)[product.id]
-
-                if product.currency_id != self.currency_id:
-                    price = product.currency_id._convert(price, self.currency_id, self.env.company, date, round=False)
+                currency = self.currency_id or currency or self.env.company.currency_id
```

Comment:
currency = currency or self.currency_id or self.env.company.currency_id

Very confusing to not use the specified param first.

Also, now that we support a currency param, I believe that we should convert the pricelist price to the given currency (if specified and different from the pricelist currency).  I know we won't need this in our code, but I believe the API would make more sense like that.

Reply:
I'd prefer not changing the order in this case, even if I can see why it's counter-intuitive. But if we have a pricelist, I think it's safe to assume that its currency should be used by default. In sale and website, if you want to change the currency, you change the pricelist to one in the wanted currency (and you don't have the currency parameter). But POS pass the currency of the POS in the parameters, but should it ever be a pricelist passed in another currency (which shouldn't be possible atm), I still think that the pricelist's currency should be used first, en not the one from the POS.

About your suggestion, this would mean that we could create pricelists without any set currency, and choose any currency to convert to afterward.... Considering my V1 for this PR (I miss the currency passed everywhere ^^), I love the idea... =D